implementations/rust/kex/src/xx.rs

```diff
@@ -90,10 +90,25 @@ impl SymmetricState {
         }
     }
 
-    pub fn new(cipher_suite: CipherSuite, vault: Arc<Mutex<dyn DynVault + Send>>) -> Self {
+    pub fn new(
+        cipher_suite: CipherSuite,
+        vault: Arc<Mutex<dyn DynVault + Send>>,
+        identity_key: Option<SecretKeyContext>,
+    ) -> Self {
+        let static_key_pair = match identity_key {
+            None => None,
+            Some(ik) => {
+                let mut v = vault.lock().unwrap();
+                let pk = v.secret_public_key_get(ik).unwrap();
```

Comment:
does this need to be handled rather than unwrapping?

Reply:
This is debatable but I think it's okay to call unwrap here.

Reply:
I especially worry about eliding the Option check here, because this code trusts the underlying Vault implementation. _You_ think it's ok because you know both the XX code and the Vault code being used here now. But in the future, it could be a different Vault implementation used, correct? In that case it could behave in a way that doesn't satisfy your current thought.

Reply:
If we don't unwrap what is the expected behavior? This requires an interface change if not unwrapped

Reply:
If the choices are:
1. unwrap and possibly panic
2. change the interface

my opinion is 2.

@mrinalwadhwa may have some thoughts?

Reply:
I have adjusted this to store a seeded value then use it later where it uses the ? operator instead of unwrapping.

Reply:
@mikelodder7 where? use Github's code linking feature's to post a comment with the link to the line where this is done please :)

Reply:
See [here](https://github.com/ockam-network/ockam/pull/518/files#diff-2b6c53777e3d5dc1c187c0864bf7c32e9811f1839d154e22f736bafa7a36f67fR134)